File: 3_private_encaptulation.py
class Student:
    def __init__(self, name, age):
        self._name = name
        self._age = age

    def set_name(self, name):
        self._name = name

    def get_name(self):
        return self._name

    def set_age(self, age):
        self._age = age

    def get_age(self):
        return self._age


student = Student("Mahbub", 20)


print("Before : ")
print("Name:", student.get_name(), ", Age:", student.get_age())
student.set_name("Ahmed")
student.set_age(22)
print("After : ")
print("Name:", student.get_name(), ", Age:", student.get_age())

File: 3_private_encaptulation.cpp
#include <bits/stdc++.h>
using namespace std;

class Student {
private:
    string name;
    int age;

public:
    Student(string n, int a) {
        name = n;
        age = a;
    }
    void setName(string n) {
        name = n;
    }
    string getName() {
        return name;
    }
    void setAge(int a) {
        age = a;
    }

    int getAge() {
        return age;
    }
};

int main() {
    Student student("Mahbub", 20);

    cout<<"Before: "<<endl;
    cout << "Name: " << student.getName() << ", Age: " << student.getAge() << endl;
    student.setName("Ahmed");
    student.setAge(22);
    cout<<"After : "<<endl;
    cout << "Name: " << student.getName() << ", Age: " << student.getAge() << endl;

    return 0;
}
